src/main/java/project/bookstore/controller/BookController.java

```diff
@@ -0,0 +1,45 @@
+package project.bookstore.controller;
+
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import project.bookstore.dto.BookDto;
+import project.bookstore.dto.CreateBookRequestDto;
+import project.bookstore.exception.EntityNotFoundException;
+import project.bookstore.mapper.BookMapper;
+import project.bookstore.model.Book;
+import project.bookstore.repository.BookRepository;
+import project.bookstore.service.BookService;
+
+@RestController
+@RequestMapping("/books")
+@RequiredArgsConstructor
+public class BookController {
+    private static final String FIND_BY_ID_MSG = "Failed to find object by id = ";
+    private final BookRepository bookRepository;
+    private final BookService bookService;
+    private final BookMapper bookMapper;
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return Optional.of(bookService.getBookById(id))
+                        .orElseThrow(() -> new EntityNotFoundException(FIND_BY_ID_MSG + id));

```

Comment:
```suggestion
        return bookService.getBookById(id);
```
you already check optional in your service layer. You don't need to do this on controller layer.